parsec/backend/asgi/rpc.py

```diff
@@ -75,3 +93,171 @@ async def anonymous_api(raw_organization_id: str):
     # Run command
     rep = await cmd_func(client_ctx, msg)
     return rpc_msgpack_rep(rep)
+
+
+rpc_auth_bp = Blueprint("authenticated_api", __name__)
+
+
+@rpc_auth_bp.route("/authenticated/<raw_organization_id>", methods=["GET", "POST"])
+async def authenticated_api(raw_organization_id: str):
+    backend: BackendApp = g.backend
+    logger = current_app.logger.getChild(authenticated_api.__name__)
+
+    logger.debug(
+        "Received call in the authenticated api for the organization {org_id} from the device={dev_id}".format(
+            org_id=raw_organization_id, dev_id=request.headers.get("Author", "Missing")
+        )
+    )
+    try:
+        organization_id = OrganizationID(raw_organization_id)
+    except ValueError:
+        abort(404)
+    try:
+        organization = await backend.organization.get(organization_id)
+    except OrganizationNotFoundError:
+        organization_exists = False
+    else:
+        organization_exists = True
+
+    # Reply to GET
+    if request.method == "GET":
+        if not organization_exists:
+            abort(404)
+        else:
+            return rpc_msgpack_rep({})
+
+    body: bytes = await request.get_data(cache=False)
+    try:
+        if request.headers["Content-Type"] != "application/msgpack":
+            raise InvalidHeaderValue("Content-Type", request.headers["Content-Type"])
+        user, device = await verify_auth_request(body, request.headers, organization, backend)
+    except (
+        IncompatibleAPIVersionsError,
+        MissingHeaderError,
+        OrganizationNotFoundError,
+        UserNotFoundError,
+        InvalidHeaderValue,
```

Comment:
Is it really necessary to bother with all those exceptions? Wouldn't it be simpler to call `abort` directly where those are raised?

Reply:
Now all of these exception are based on the same root exception `HttpProtocolError`, so this discussion is outdated